SVGEngine.mm

```diff
@@ -226,9 +259,14 @@ @interface SVGAttributeSet () {
         const char * const attrName  = (char *)xmlTextReaderConstName(_xmlReader),
                    * const attrValue = (char *)xmlTextReaderConstValue(_xmlReader);
 
-        if(strcasecmp("style", attrName) == 0)
-            [attrs addEntriesFromDictionary:_SVGParseStyle(@(attrValue))];
-        else if(strcasecmp("transform", attrName) == 0) {
+        if(strcasecmp("style", attrName) == 0){
+            NSMutableDictionary *style = _SVGParseStyle(@(attrValue));
+            //Don't allow overriding of display:none
+            if ([attrs[@"display"] isEqualToString:@"none"] && style[@"display"] != nil) {
```

Comment:
That second check feels redundant (if the key isn't present, the next line just does nothing if I recall correctly)

Also the string comparison should probably be case insensitive.